scripts/speech_recognition/code_switching/code_switching_manifest_creation.py

```diff
@@ -0,0 +1,175 @@
+        max_sample_duration_sec: Maximum permissible duration of generated CS sample in sec
+        min_sample_duration_sec: Minimum permissible duration of generated CS sample in sec
+        data_requirement_hrs: Required size of generated corpus
+
+    Returns:
+        Created synthetic CS manifest as list
+
+    """
+
+    total_duration = 0
+    constructed_data = []
+
+    num_samples_lang0 = len(data_lang_0)
+    num_samples_lang1 = len(data_lang_1)
+
+    while total_duration < (data_requirement_hrs * 3600):
+
+        created_sample_duration_sec = 0
+        created_sample_dict = {}
+        created_sample_dict['lang_ids'] = []
+        created_sample_dict['texts'] = []
+        created_sample_dict['paths'] = []
+        created_sample_dict['durations'] = []
+
+        while created_sample_duration_sec < min_sample_duration_sec:
+
+            lang_selection = random.randint(0, 1)
+
+            if lang_selection == 0:
+                index = random.randint(0, num_samples_lang0 - 1)
+                sample = data_lang_0[index]
+                lang_id = lid_lang_0
+            else:
+                index = random.randint(0, num_samples_lang1 - 1)
+                sample = data_lang_1[index]
+                lang_id = lid_lang_1
+
+            if (created_sample_duration_sec + sample['duration']) > max_sample_duration_sec:
+                continue
+            else:
+                created_sample_duration_sec += sample['duration']
+                created_sample_dict['lang_ids'].append(lang_id)
+                created_sample_dict['texts'].append(sample['text'])
+                created_sample_dict['paths'].append(sample['audio_filepath'])
+                created_sample_dict['durations'].append(sample['duration'])
+
+        created_sample_dict['total_duration'] = created_sample_duration_sec
+        constructed_data.append(created_sample_dict)
+        total_duration += created_sample_duration_sec
+
+    return constructed_data
+
+
+def main():
+
+    manifest0 = args.manifest_language1
+    manifest1 = args.manifest_language2
+    lid0 = args.id_language1
+    lid1 = args.id_language2
+    min_sample_duration = args.min_sample_duration_sec
+    max_sample_duration = args.max_sample_duration_sec
+    dataset_requirement = args.dataset_size_required_hrs
+    manifest_save_path = args.manifest_save_path
+
+    # Reading data
+    logging.info('Reading manifests')
+    data_language0 = __read_manifest(manifest0)
+    data_language1 = __read_manifest(manifest1)
+
+    # Creating the CS data Manifest
+    logging.info('Creating CS manifest')
+    constructed_data = __create_cs_manifest(
+        data_language0, data_language1, lid0, lid1, max_sample_duration, min_sample_duration, dataset_requirement
+    )
+
+    # Saving Manifest
+    logging.info('saving manifest')
+    __write_manifest(manifest_save_path, constructed_data)
+
+    print("Synthetic CS manifest saved at :", manifest_save_path)
```

Comment:
might be worth printing some stats at the end of the scripts -- e.g. total # of samples from each language and the coverage, etc.. ?